actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -19,35 +19,36 @@ module ActionController
   #   params.require(:a)
   #   # => ActionController::ParameterMissing: param is missing or the value is empty: a
   class ParameterMissing < KeyError
-    attr_reader :param, :keys # :nodoc:
-
-    def initialize(param, keys = nil) # :nodoc:
-      @param = param
-      @keys  = keys
-      super("param is missing or the value is empty: #{param}")
-    end
-
-    class Correction
-      def initialize(error)
-        @error = error
+    # In Ruby <= 2.5 +KeyError#new+ does not take the +key+ or +receiver+
+    # arguments and we need this feature detection.
+    takes_key_and_receiver = begin
+                               !!new(nil, key: nil, receiver: nil)
+                             rescue ArgumentError
+                               false
+                             end
+
+    if takes_key_and_receiver
+      attr_reader :param # :nodoc:
+
+      def initialize(param, receiver: nil) # :nodoc:
+        @param = param
+        super("param is missing or the value is empty: #{param}", key: param, receiver: receiver)
       end
-
-      def corrections
-        if @error.param && @error.keys
-          maybe_these = @error.keys
-
-          maybe_these.sort_by { |n|
-            DidYouMean::Jaro.distance(@error.param.to_s, n)
-          }.reverse.first(4)
-        else
-          []
-        end
+    else
+      attr_reader :param, :key, :receiver # :nodoc:
```

Comment:
```suggestion
      attr_reader :key, :receiver # :nodoc:
      alias_method :param, :key:
```

When Ruby 2.5 support is dropped, we can remove the first line.